File: src/main/java/com/jay/model/Player.java
package com.jay.model;

import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;


@Entity
@Table(name = "Player")
public class Player {
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE)
	@Column(name = "player_id")
	private int id;
	@NotNull
	@Column(unique = true)
	private String username;
	@NotNull
	private String firstname;
	@NotNull
	private String lastname;
	private int draws;
	private int wins;
	private int winStreaks;
	private int winStreakCounter;
	private int loses;
	private int lossStreaks;
	private int lossStreakCounter;
	private int gotFullBalled;
	private int fullBalledEnemy;
	
	@ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
	@JoinTable
			(
			name = "player_match",
			joinColumns = @JoinColumn(name = "player_id", referencedColumnName = "player_id"),
			inverseJoinColumns = @JoinColumn(name = "match_id", referencedColumnName = "match_id")
			)
	private Set<Match> matches;

	public Player() {
		
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getFirstname() {
		return firstname;
	}

	public void setFirstname(String firstname) {
		this.firstname = firstname;
	}

	public String getLastname() {
		return lastname;
	}

	public void setLastname(String lastname) {
		this.lastname = lastname;
	}

	public int getDraws() {
		return draws;
	}

	public void setDraws(int draws) {
		this.draws = draws;
	}

	public int getWins() {
		return wins;
	}

	public void setWins(int wins) {
		this.wins = wins;
	}

	public int getWinStreaks() {
		return winStreaks;
	}

	public void setWinStreaks(int winStreaks) {
		this.winStreaks = winStreaks;
	}

	public int getWinStreakCounter() {
		return winStreakCounter;
	}

	public void setWinStreakCounter(int winStreakCounter) {
		this.winStreakCounter = winStreakCounter;
	}

	public int getLoses() {
		return loses;
	}

	public void setLoses(int loses) {
		this.loses = loses;
	}

	public int getLossStreaks() {
		return lossStreaks;
	}

	public void setLossStreaks(int lossStreaks) {
		this.lossStreaks = lossStreaks;
	}

	public int getLossStreakCounter() {
		return lossStreakCounter;
	}

	public void setLossStreakCounter(int lossStreakCounter) {
		this.lossStreakCounter = lossStreakCounter;
	}

	public int getGotFullBalled() {
		return gotFullBalled;
	}

	public void setGotFullBalled(int gotFullBalled) {
		this.gotFullBalled = gotFullBalled;
	}

	public int getFullBalledEnemy() {
		return fullBalledEnemy;
	}

	public void setFullBalledEnemy(int fullBalledEnemy) {
		this.fullBalledEnemy = fullBalledEnemy;
	}

	public int getId() {
		return id;
	}
	
	public Set<Match> addMatch(Match match) {
		matches.add(match);
		return matches;
	}
	
	public Set<Match> removeMatch(Match match) {
		matches.remove(match);
		return matches;
	}
	

}

File: src/main/webapp/js/player.js
class Player {
    id;
    username;
    firstname;
    lastname;
    wins = 0;
    winStreaks = 0;
    winStreakCounter = 0;
    loses = 0;
    lossStreaks = 0;
    lossStreakCounter = 0;
    draws = 0;
    fullBalledEnemy = 0;
    gotFullBalled = 0;

    constructor(username, firstname, lastname) {
        this.username = username;
        this.firstname = firstname;
        this.lastname = lastname;
    }

    get getId() {
        return this.id;
    }

    get getUsername() {
        return this.username;
    }

    set setFirstname(value) {
        this.firstname = value;
    }

    get getFirstname() {
        return this.firstname;
    }

    set setLastname(value) {
        this.lastname = value;
    }

    get getLastname() {
        return this.lastname;
    }

    addWin() {
        this.wins++;
        this.winStreakCounter++;
        if (this.winStreakCounter > this.winStreaks) {
            this.winStreaks++;
            this.lossStreakCounter = 0;
        }
    }
    set setWins(value) {
        this.wins = value;
    }

    get getWins() {
        return this.wins;
    }

    set setWinStreaks(value) {
        this.winStreaks = value;
    }

    get getWinStreaks() {
        return this.winStreaks;
    }

    get getWinStreakCounter() {
        return this.winStreakCounter
    }

    set setWinStreakCounter(value) {
        this.winStreakCounter = value;
    }
    addLoss() {
        this.loses++;
        this.lossStreakCounter++;
        if (this.lossStreakCounter > this.lossStreaks) {
            this.lossStreaks++;
            this.winStreakCounter = 0;
        }
    }

    set setLoss(value) {
        this.loses = value;
    }

    get getLoss() {
        return this.loses;
    }

    set setLossStreaks(value) {
        this.lossStreaks = value;
    }

    get getLossStreaks() {
        return this.lossStreaks;
    }

    set setLossStreakCounter(value) {
        this.lossStreakCounter = vale;
    }

    get getLossStreakCounter() {
        return this.lossStreakCounter;
    }

    set setDraws(value) {
        this.draws = value;
    }

    get getDraws() {
        return this.draws;
    }

    get getWinStreaks() {
        return this.winStreaks;
    }

    get getLossesStreak() {
        return this.lossStreaks;
    }

    set setFullBalledEnemy(value) {
        this.fullBalledEnemy = value;
    }

    get getFullBalledEnemy() {
        return this.fullBalledEnemy;
    }

    set setGotFullBalled(value) {
        this.gotFullBalled = value;
    }

    get getGotFullBalled() {
        return this.gotFullBalled;
    }
}